workflow_plandefinition/models/workflow_plan_definition_action.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields, models, api, exceptions
+
+
+class PlanDefinitionAction(models.Model):
+    _name = 'workflow.plan.definition.action'
+    _description = 'Medical Plan Definition Action'
+    _inherit = 'mail.thread'
+    _parent_name = 'parent_id'
+    _parent_store = True
+    _parent_order = 'name'
+    _order = 'parent_left'
+    _rec_name = 'complete_name'
+    _order = 'id asc'
+
+    @api.one
+    @api.depends('name', 'parent_id')
+    def _compute_complete_name(self):
+        """ Forms complete name of action from parent to child action. """
+        name = self.name
+        current = self
+        while current.parent_id:
+            current = current.parent_id
+            name = '%s/%s' % (current.name, name)
+        self.complete_name = name
+
+    @api.multi
+    @api.depends('parent_id', 'direct_plan_definition_id')
+    def _compute_plan_definition_id(self):
+        for rec in self:
+            rec.plan_definition_id = rec.direct_plan_definition_id
+            if rec.parent_id:
+                rec.plan_definition_id = rec.parent_id.plan_definition_id
```

Comment:
We're not trying to account for recursion here right? As in- we don't want the parent's parent's plan_definition_id if it's defined? That's currently the case, just making sure.

Reply:
No, we do not want to use recursion. I'm not sure if I understood your point, but the reason why we do this is for every action to know to which plan definition it belongs to. As the plan definition itself is not a parent_id, then it had to be passed like that. Is that what you mean?

Reply:
Woot thanks for the exlanation @rgarnau - makes sense now!